### ISSUE TITLE
Introduce train/test split

### ISSUE THREAD

**Issue by manoj-bhaskaran** [manoj-bhaskaran/expense-predictor#42]
```python
# model_runner.py:109-115
model.fit(X_train, y_train)
y_train_predictor = model.predict(X_train)  # Predicting on training data!

rmse = np.sqrt(mean_squared_error(y_train, y_train_predictor))
```

**Problem:**
- Models are evaluated on training data only
- No validation set or test set
- Reported metrics don't reflect true generalization performance
- Risk of overfitting not detected

Introduce a train/test split to resolve this problem.
